src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -396,7 +396,8 @@ private bool TryGetWordExtent(CompletionParams request, LSPDocumentSnapshot docu
                 var baseIndentation = Math.Max(GetBaseIndentation(wordExtent, formattingOptions) - 1, 0);
                 completionList = IncludeCSharpSnippets(baseIndentation, completionList, formattingOptions);
             }
-            else if (IsWordOnEmptyLine(wordExtent, documentSnapshot))
+            //if all completion items are properties then completion is requested inside initializer syntax and we don't need to add snippets
+            else if (IsWordOnEmptyLine(wordExtent, documentSnapshot) && !completionList.Items.All(el => el.Kind == CompletionItemKind.Property))
```

Comment:
Done. But what is the performance boost here? I found implementation of `All` method and it seems to be almost the same, as you suggested: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/AnyAll.cs#L70-L91. Does the single allocation of `Func` really matter here? Or `foreach` is slower than `for`? As I know, Linq is bad for performance when using "chain" methods like `Select`, `Where` etc. When the method returns an actual result like `Any` or `All` it does not drop performance. Am I wrong here?

Reply:
Thanks @DoctorKrolic!

> Does the single allocation of Func really matter here?

So in practice not really. Also FWIW I believe the compiler will actually optimize away the Func allocation in this scenario because it's not capturing anything (aka it'll generate a dummy method stub for it); for the future you can ensure this by putting `static` in front of the lambda creation for your own code. That being said just because it doesn't currently allocate doesn't necessarily mean it's great. Reason being, if we start unintentionally referencing a local member in the class within the lambda it then ends up capturing and does result in allocating a Func every time which becomes worse. 

> Or foreach is slower than for

This is the main issue one but it's also minimal gains. Whenever you do `.All` it has to allocate an enumerator and on hot paths those enumerators can really add up. Now that being said the compiler can also be smarter in some situations where if it recognizes that the type itself is an Array or List it'll actually translate `foreach`s into `for`s on your behalf.

> When the method returns an actual result like Any or All it does not drop performance.

Enumerators still have to be allocated 😄 

-------

The main motivation behind changing from `.All` wasn't for perf but more for consistency across the codebase with a little side of extra resilient for future changers. Oh and because it's on the completion path it makes the bar a little higher for changes.